README.md
# emacs.d

emacs stuffs

easyscripts/news
#!/usr/bin/env python3
import requests
import sys
import re, base64
import subprocess
from bs4 import BeautifulSoup


def extractLinks(TEXT):
    # Decode only valid Links
    try:
        # Extract only the base64 string
        _temp1 = re.split('/', TEXT)
        _temp2 = _temp1[-1]+"?"
        _temp2 = _temp2.split('?')[0]
        # Decode & Add padding
        decodeStr = str(base64.b64decode(_temp2 + "=="))
        # Regex find full url
        XX = re.findall(r'http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\), ]|(?:%[0-9a-fA-F][0-9a-fA-F]))+', decodeStr)
        # Get the news link
        return XX[0].split('\\')[0]
    except Exception:
        pass

def createHTMLPage(NEWS, PAGE="/tmp/news.html"):
    F = open(PAGE, "w")
    F.writelines(f"<h2 style='color:#aaa'> <center> Query: {' '.join(sys.argv[1:])}  |  {len(NEWS)} Stories Found </center></h2><br><br>")
    F.writelines("<style> body{background-color:#1a1c23} </style><body>")
    indexN=0
    for n in NEWS:
        F.write(f"""<div style='background-color:#1e2028;border-radius:8px;padding:5px;'><center><a href='{n[-1]}' target='_blank' style='color:#aaa;'><h2><span style="float:left;color:#606060;font-size:13px;">[ {indexN} ]</span><span style="float:left;margin-left:2%;color:red;font-size:19px;">{n[2]}</span>{n[0]}<span style="float:right;margin-right:8%;color:#aaa;font-size:19px;">{n[1]}</span></h2></a></center></div><br>""")
        indexN+=1
    F.writelines("</body>")
    subprocess.run(['xdg-open', PAGE])

def termRender(Data, Line=10, titleLen=75):
    n = 0
    indexUrl = []
    for i in Data:
        # Align String
        titleFormat = i[0][:titleLen]
        if len(titleFormat) < titleLen:
            spaceToAdd = " "*(titleLen - len(titleFormat))
            titleFormat = titleFormat+spaceToAdd
        # Render Info
        print(f"[ {n} ] {titleFormat : ^15}\t{i[2]}\t{i[1]}")

        indexUrl.append(i[-1])
        n += 1

        if n % Line == 0:
            try:
                choose = list(input(" [ q to Quit ]   [ Enter for More ] \n"))
                choose = [i for i in choose if i != " "]

                if len(choose) == 0: pass
                elif choose[0] == "q": exit()                
                
                if len(choose) > 1:
                    for c in choose: subprocess.run(['xdg-open', indexUrl[int(c)]])
                elif len(choose) == 1:
                    subprocess.run(['xdg-open', indexUrl[int(choose[0])]])
                    
                indexUrl.clear()
                n = 0
                
            except Exception as e:
                print('Error', e)
                exit()

def newToFile(dataArr, numOfStories=50, FILE="/home/ss/.emacs.d/easyscripts/news.data"):
    ''' Get News Data and Write to a file with seperator '''
    try:
        with open(FILE, "w") as F:            
            for d in dataArr[:numOfStories]:
                if d[-1] is None: d[-1] = ""    # Replace None
                Stories = "||".join(d)          # Join with '||' seperator                
                F.write(f"{Stories}\n")        #  Write to a file
                
    except Exception as e:
        print(e)
        pass

                    
class GNews:
    def __init__(self, Topic):        
        self.Url = f"https://news.google.com/search?q={Topic}&hl=en-US&gl=US&ceid=US%3Aen"
        
    def News(self):
        try:
            s = requests.session()
            headers = {'User-Agent': 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:111.0)'\
                       'Gecko/20100101 Firefox/111.0',
                       'Accept': 'text/html,application/xhtml+xml,application/xml;'\
                       'q=0.9,image/webp,*/*;q=0.8'}
            cookies = {'CONSENT': 'YES+'}

            r = s.get(self.Url, headers=headers, cookies=cookies)
            # Return HTML Data
            return r.text
        except Exception as e:
            print(str(e))

    def getNews(self, toSort=True):
        '''
        Use Html parser to get all the needed Data
        Returns a list news storys
        '''
        soup = BeautifulSoup(self.News(), features="html.parser")        
        link = [extractLinks(i['href']) for i in soup.find_all('a', class_="VDXfz")]
        title = [i.text for i in soup.find_all('h3', class_="ipQwMb")]
        source = [i.text for i in soup.find_all('div', class_="wsLqz")]
        time = [i.text for i in soup.find_all('time', class_="WW6dff")]        

        dicData = {"AllNews":[]}
        for i in range(len(link)): dicData["AllNews"].append([title[i], source[i], time[i], link[i]])
        if toSort:
            # Minutes, Hours, Days old News stories
            all_d, list_d, list_hr, list_min = [],[],[],[]
            for i in dicData["AllNews"]:
                if "minute" in i[2] or "minutes" in i[2]: list_min.append(i)
                elif "hour" in i[2] or "hours" in i[2]: list_hr.append(i)
                elif "Yesterday" in i[2] or "yesterday" in i[2]: list_d.append(i)
                else: all_d.append(i)
            # Sorted Recent
            return list_min + list_hr + list_d + all_d
        # Unsorted
        return dicData["AllNews"]
    
        
if __name__ == '__main__':
    import argparse

    parser = argparse.ArgumentParser()
    parser.add_argument("topic", help="Topic to Search", nargs='+')
    parser.add_argument("-t", help="Output in Terminal", action="store_true")
    parser.add_argument("-html", help="Output as Html file", action="store_true")    
    
    args = parser.parse_args()
    topic = "-".join(args.topic)    
    N = GNews(topic).getNews()

    print(len(N))
    
    if args.t: termRender(N)
    elif args.html: createHTMLPage(N)
    else: newToFile(N)


init.el
;; [ Emacs Config ]


(require 'package)
(setq package-enable-at-startup nil)
(add-to-list 'package-archives '("melpa" . "https://melpa.org/packages/"))
(package-initialize)

(org-babel-load-file (expand-file-name "~/.emacs.d/myInit.org"))


(custom-set-variables
 ;; custom-set-variables was added by Custom.
 ;; If you edit it by hand, you could mess it up, so be careful.
 ;; Your init file should contain only one such instance.
 ;; If there is more than one, they won't work right.
 '(auth-source-save-behavior nil)
 '(blink-matching-paren t)
 '(centered-window-mode t)
 '(custom-enabled-themes '(firebelly))
 '(custom-safe-themes '(default))
 '(cwm-centered-window-width 150)
 '(dashboard-banner-logo-title nil)
 '(dashboard-bookmarks-item-format "%s")
 '(dashboard-center-content t)
 '(dashboard-filter-agenda-entry 'dashboard-no-filter-agenda)
 '(dashboard-footer "Who the hell uses VIM anyway? Go Evil!")
 '(dashboard-icon-type nil)
 '(dashboard-set-file-icons nil)
 '(dashboard-set-footer nil)
 '(dashboard-set-heading-icons t)
 '(dashboard-set-navigator t)
 '(dashboard-show-shortcuts nil)
 '(initial-frame-alist '((fullscreen . maximized)))
 '(package-selected-packages
   '(rust-mode vterm magit simple-modeline swiper pyvenv bongo pdf-tools dired-subtree mood-line yasnippet vertico projectile org-bullets multiple-cursors elfeed dashboard company all-the-icons-dired all-the-icons-completion))
 '(pdf-view-midnight-colors '("gray60" . "#101010") t)
 '(pdf-view-midnight-invert nil t)
 '(tab-bar-close-button-show nil)
 '(tab-bar-format
   '(tab-bar-format-history tab-bar-format-tabs tab-bar-separator))
 '(tab-bar-mode t)
 '(tab-line-new-button-show nil)
 '(vertico-mode t))


(custom-set-faces
 ;; custom-set-faces was added by Custom.
 ;; If you edit it by hand, you could mess it up, so be careful.
 ;; Your init file should contain only one such instance.
 ;; If there is more than one, they won't work right.
 '(dashboard-navigator ((t (:foreground "grey 30"))))
 '(elfeed-search-title-face ((t (:foreground "gray25" :underline "gray30" :height 1.3))))
 '(elfeed-search-unread-title-face ((t (:foreground "gray50" :weight bold))))
 '(flymake-warning ((t nil)))
 '(fringe ((t (:background "#101010"))))
 '(tab-bar-tab ((t (:background "#151515" :foreground "#bbbbbb"))))
 '(tab-bar-tab-group-current ((t (:inherit tab-bar-tab :weight bold))))
 '(tab-bar-tab-inactive ((t (:background "#303030" :foreground "gray55" :height 0.8))))
 '(vertico-current ((t (:inherit highlight :extend t :background "#303030" :foreground "SeaGreen2" :weight semi-bold))))
 '(yafolding-ellipsis-face ((t nil)) t))




lisp/yafolding.el
;;; yafolding.el --- Folding code blocks based on indentation

;; Copyright (C) 2013-2017 Zeno Zeng

;; Author: Zeno Zeng <zenoofzeng@gmail.com>
;; Keywords: folding
;; Package-Version: 20200119.1353
;; Package-Commit: 4c1888ae45f9241516519ae0ae3a899f2efa05ba

;; This program is free software; you can redistribute it and/or modify
;; it under the terms of the GNU General Public License as published by
;; the Free Software Foundation, either version 3 of the License, or
;; (at your option) any later version.

;; This program is distributed in the hope that it will be useful,
;; but WITHOUT ANY WARRANTY; without even the implied warranty of
;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
;; GNU General Public License for more details.

;; You should have received a copy of the GNU General Public License
;; along with this program.  If not, see <http://www.gnu.org/licenses/>.

;;; Commentary:

;; Folding code blocks based on indentation.
;;; Code:

(defgroup yafolding nil
  "Fold code blocks based on indentation level"
  :prefix "yafolding-"
  :link '(url-link :tag "yafolding on github" "https://github.com/zenozeng/yafolding.el")
  :group 'applications)

(defface yafolding-ellipsis-face
  '()
  "Face for folded blocks"
  :group 'yafolding)

(defcustom yafolding-ellipsis-content "..."
  "Text to show in place of a folded block."
  :tag "Ellipsis"
  :type 'string
  :group 'yafolding
  )

(defcustom yafolding-show-fringe-marks t
  "Show fold markers in the fringe?"
  :tag "Show fringe marks?"
  :type 'boolean
  :group 'yafolding)

(defun yafolding-get-overlays (beg end)
  "Get all overlays between BEG and END."
  (delq nil
        (mapcar (lambda (overlay)
                  (and (member "yafolding" (overlay-properties overlay))
                       overlay))
                (overlays-in beg end))))

(defun yafolding-should-ignore-current-line-p ()
  "Return if should ignore current line."
  (string-match-p "^[ \t]*$"
                  (buffer-substring-no-properties
                   (line-beginning-position)
                   (line-end-position))))

(defun yafolding-get-indent-level ()
  "Get the indent level of current line."
  (interactive)
  (if (and (yafolding-should-ignore-current-line-p)
           (< (line-number-at-pos) (line-number-at-pos (point-max))))
      (save-excursion
        (forward-line 1)
        (yafolding-get-indent-level))
      (let ((indent-level 0)
            (last-indentation (current-indentation)))
        (save-excursion
          (while (and (> (current-indentation) 0)
                      (> (line-number-at-pos) 1))
            (forward-line -1)
            (when (< (current-indentation) last-indentation)
              (setq last-indentation (current-indentation))
              (setq indent-level (+ 1 indent-level)))))
        indent-level)))

(defun yafolding-show-region (beg end)
  "Delete all yafolding overlays between BEG and END."
  (mapcar 'delete-overlay (yafolding-get-overlays beg end)))

(defun yafolding-show-all ()
  "Delete all yafolding overlays."
  (interactive)
  (yafolding-show-region (point-min) (point-max)))

(defun yafolding-hide-all (&optional indent-level)
  "Hide all elements based on INDENT-LEVEL."
  (interactive)
  (unless indent-level
    (setq indent-level (yafolding-get-indent-level)))
  (save-excursion
    (goto-char (point-min))
    (while (< (line-number-at-pos)
            (line-number-at-pos (point-max)))
      (if (and (= (yafolding-get-indent-level) indent-level)
             (not (yafolding-should-ignore-current-line-p)))
        (yafolding-hide-element))
    (forward-line 1))))

(defun yafolding-toggle-all (&optional indent-level)
  "Toggle folding of the entire file.

If given, toggle all entries that start at INDENT-LEVEL."
  (interactive)
  (unless indent-level
    (setq indent-level (yafolding-get-indent-level)))
  (if (yafolding-get-overlays (point-min) (point-max))
      (yafolding-show-all)
    (yafolding-hide-all indent-level)))

(defun yafolding-ellipsis ()
  "Return propertized ellipsis content."
  (concat " "
          (propertize yafolding-ellipsis-content 'face 'yafolding-ellipsis-face)
          " "))

(defun yafolding-hide-region (beg end)
  "Hide region between BEG and END."
  (when (> end beg)
      (yafolding-show-region beg end)
      (let ((before-string
             (concat
              (when yafolding-show-fringe-marks
                (propertize " " 'display '(left-fringe right-triangle)))
              (yafolding-ellipsis)))
            (new-overlay (make-overlay beg end)))
        (overlay-put new-overlay 'invisible t)
        (overlay-put new-overlay 'intangible t)
        (overlay-put new-overlay 'evaporate t)
        (overlay-put new-overlay 'modification-hooks
                     (list (lambda (overlay &optional a b c d)
                             (delete-overlay overlay))))
        (overlay-put new-overlay 'before-string before-string)
        (overlay-put new-overlay 'category "yafolding"))))

(defun yafolding-debug ()
  "Show yafolding information of the current position."
  (interactive)
  (message "indentation: %d, indent level: %d, ingore current line: %s, element-region: %d - %d, (L%d - L%d)"
           (yafolding--current-indentation)
           (yafolding-get-indent-level)
           (yafolding-should-ignore-current-line-p)
           (car (yafolding-get-element-region))
           (car (cdr (yafolding-get-element-region)))
           (line-number-at-pos (car (yafolding-get-element-region)))
           (line-number-at-pos (car (cdr (yafolding-get-element-region))))))

(defun yafolding-get-element-region ()
  "Get '(beg end) of current element."
  (let ((beg (line-end-position))
        (end (line-end-position))
        (indentation (current-indentation)))
    (save-excursion
      (next-line)
      (while (and (< (line-number-at-pos) (line-number-at-pos (point-max)))
                  (or (> (current-indentation) indentation)
                      (yafolding-should-ignore-current-line-p)))
        (unless (yafolding-should-ignore-current-line-p)
          (setq end (line-end-position)))
        (next-line))) ; using next-line instead of forward-line, for issue#23
    (list beg end)))

(defun yafolding-hide-element ()
  "Hide current element."
  (interactive)
  (let ((region (yafolding-get-element-region)))
    (yafolding-hide-region (car region)
                           (cadr region))))

(defun yafolding-show-element ()
  "Show current element."
  (interactive)
  (yafolding-show-region (line-beginning-position)
                         (+ 1 (line-end-position))))

(defun yafolding-toggle-element ()
  "Toggle current element."
  (interactive)
  (if (yafolding-get-overlays (line-beginning-position)
                              (+ 1 (line-end-position)))
      (yafolding-show-element)
    (yafolding-hide-element)))

(add-hook 'isearch-mode-hook
          (lambda() (mapcar (lambda (overlay)
                          (overlay-put overlay 'invisible nil))
                        (yafolding-get-overlays (point-min) (point-max)))))

(add-hook 'isearch-mode-end-hook
          (lambda() (mapcar (lambda (overlay)
                          (overlay-put overlay 'invisible t))
                        (yafolding-get-overlays (point-min) (point-max)))))

(defun yafolding-go-parent-element ()
  "Go back to parent element."
  (interactive)
  (re-search-backward (concat "^.\\{,"
                              (number-to-string (- (current-indentation) 1))
                              "\\}[^ \t]+")))

(defun yafolding-hide-parent-element ()
  "Hide the parent element."
  (interactive)
  (ignore-errors
    (yafolding-go-parent-element)
    (yafolding-hide-element)))

;;;###autoload
(defvar yafolding-mode-map
  (let ((map (make-sparse-keymap)))
    (define-key map (kbd "<C-S-return>") #'yafolding-hide-parent-element)
    (define-key map (kbd "<C-M-return>") #'yafolding-toggle-all)
    (define-key map (kbd "<C-return>") #'yafolding-toggle-element)
    map))

;;;###autoload
(define-minor-mode yafolding-mode
  "Toggle yafolding mode."
  :keymap yafolding-mode-map)

(provide 'yafolding)
;;; yafolding.el ends here


firebelly/firebelly-theme.el
;;  [ Customize ] firebelly-theme.el
;; Original Author: startling
;; URL: https://github.com/startling/firebelly
;; Version: 1.0
;; Package-Requires: ((cl-lib "0.5"))

;;;###autoload
(when load-file-name
  (add-to-list 'custom-theme-load-path
               (file-name-as-directory
               (file-name-directory load-file-name))))

(require 'cl-lib)

(deftheme firebelly)
(provide-theme 'firebelly)
(let
  ;; This beautiful palette is shamelessly stolen from chris
    ;; kempson's base16: https://github.com/chriskempson/base16
    ;; 90a959
  ((palette
    '("#aa4c4d" ; RED
      "#d28445"  ;ORANGE
      "#f4bf75" ; YELLOW
      "#6798DB" ; GREEN 
      "#75b5aa" ; LIGHT BLUE
      "#699159" ; 
      "#aa759f" ; PURPLE
      "#8f5536" ; LIGHT RED
      "#d9d9d9"))
   ;; Simple grayscale palette.
   (greys
    '("#101010" "#292929" "#954141" "#555555"
      "#666666" "#6f6f6f" "#969696" "#999999" "#606060")))

  ;; Tiny functions to make getting from the palettes easier.
  (cl-flet
      ((color (n) (nth n palette))
       (grey (n) (nth n greys)))

    ;; Define some faces for our theme.
    (custom-theme-set-faces
     'firebelly
    ;;;; Styling emacs.
     `(default
       ((t (:background ,(grey 0) :foreground ,(grey 7)))))
     `(cursor
       ((t (:background , "#ffffff"))))
     `(highlight
       ((t (:background ,(grey 0)))))
     `(shadow
       ((t (:foreground ,(grey 6)))))
     `(isearch
       ((t (:background ,(grey 2) :foreground ,(color 6)))))
     `(query-replace
       ((t (:background ,(grey 2) :foreground ,(color 6)))))
     `(lazy-highlight
       ((t (:background ,(grey 1) :foreground ,(grey 3)))))
     `(minibuffer-prompt
       ((t (:foreground ,(color 1)))))
     `(trailing-whitespace
       ((t (:background ,(grey 1)))))
     `(nobreak-space
       ((t (:background ,(grey 1)))))
     `(escape-glyph
       ((t (:foreground ,(color 2)))))
     ;; Fringes are ugly.
     `(fringe
       ((t (:background ,(grey 0)))))
     ;; Highlight the border.
     `(vertical-border
       ((t (:foreground ,(grey 1)))))
     ;; Mode lines look the same but for the text.
     `(mode-line
       ((t (:background ,(grey 0) :foreground ,(grey 6) :box nil))))
     `(mode-line-inactive
       ((t (:background ,(grey 1) :foreground ,(grey 3) :box nil))))
     `(header-line
       ((t (:background ,(grey 1) :foreground ,(grey 3) :box nil))))
     `(mode-line-buffer-id
       ((t (:bold t))))
     `(mode-line-highlight
       ((t (:foreground ,(grey 7)))))
     ;; Regions are slightly lighter.
     `(region
       ((t (:background ,(grey 3)))))
     `(secondary-selection
       ((t (:background ,(grey 2)))))

    ;;;; Styling code.
     ;; Comments are lighter than their delimiters.
     `(font-lock-comment-face
       ((t (:foreground ,(grey 5) :slant italic))))
     `(font-lock-comment-delimiter-face
       ((t (:foreground ,(grey 1)))))
     ;; Docstrings are slightly lighter.
     `(font-lock-doc-face
       ((t (:foreground ,(grey 4) :background ,(grey 0)))))
     `(font-lock-function-name-face
       ((t (:foreground ,(color 3)))))
     `(font-lock-variable-name-face
       ((t (:foreground , (color 0)))))
     `(font-lock-builtin-face
       ((t (:foreground ,(color 6)))))
     `(font-lock-constant-face
       ((t (:foreground ,(color 1)))))
     `(font-lock-type-face
       ((t (:foreground ,(color 4)))))
     `(font-lock-string-face
       ((t (:foreground ,(color 5) :background ,(grey 1) :slant italic))))
     `(font-lock-keyword-face
       ((t (:foreground ,(color 6)))))

    ;;;; Styling extensions.
     ;; Erc faces.
     `(erc-notice-face
       ((t (:foreground ,(grey 2)))))
     `(erc-current-nick-face
       ((t (:foreground ,(color 3) :bold t))))
     `(erc-prompt-face
       ((t (:foreground ,(color 1) :background ,(grey 0)))))
     `(erc-input-face
       ((t (:foreground ,(grey 6)))))
     `(erc-my-nick-face
       ((t (:foreground ,(color 1)))))
     `(erc-error-face
       ((t (:foreground ,(color 0)))))
     `(erc-timestamp-face
       ((t (:foreground ,(color 3)))))
     ;; Rainbow delimiters are mostly in order.
     `(rainbow-delimiters-depth-1-face
       ((t (:foreground ,(color 7)))))
     `(rainbow-delimiters-depth-2-face
       ((t (:foreground ,(color 6)))))
     `(rainbow-delimiters-depth-3-face
       ((t (:foreground ,(color 5)))))
     `(rainbow-delimiters-depth-4-face
       ((t (:foreground ,(grey 3)))))
     `(rainbow-delimiters-depth-5-face
       ((t (:foreground ,(color 4)))))
     `(rainbow-delimiters-depth-6-face
       ((t (:foreground ,(color 3)))))
     `(rainbow-delimiters-depth-7-face
       ((t (:foreground ,(color 2)))))
     `(rainbow-delimiters-depth-8-face
       ((t (:foreground ,(color 1)))))
     `(rainbow-delimiters-depth-9-face
       ((t (:foreground ,(color 0)))))
     ;; shm faces
     `(shm-current-face
       ((t (:background ,(grey 1)))))
     `(shm-quarantine-face
       ((t (:foreground ,(color 0) :background ,(grey 1)))))
     ;; linum-relative faces
     `(linum-relative-current-face
       ((t (:background ,(grey 0) :foreground ,(color 1)))))
     ; Helm
     `(helm-selection
       ((t (:foreground, "light grey" :background ,"#303030" :bold t))))

   ;; tab-bar-mode
     `(tab-bar-tab-inactive
       ((t (:background, "#000000" :foreground, "#6e6e6e"))))
     `(tab-bar-tab
       ((t (:background, "#404040" :foreground, "#b8b8b8"))))
     `(tab-bar
       ((t (:background, "#151515" :height 1.2))))

     ; Company
     '(company-echo-common ((t (:foreground "SpringGreen3"))))
     '(company-preview-common ((t (:inherit company-preview :foreground "SpringGreen3"))))
     '(company-scrollbar-fg ((t (:background "#202020"))))
     '(company-template-field ((t (:background "gray20"))))
     '(company-tooltip ((t (:background "#151515" :foreground "gray50"))))
     '(company-tooltip-annotation ((t (:foreground "gray60"))))
     '(company-tooltip-common ((t (:foreground "white smoke"))))
     '(company-tooltip-common-selection ((t (:inherit company-tooltip-common))))
     '(company-tooltip-selection ((t (:background "#404040"))))
     
     ; Term
     '(term-color-black ((t (:foreground "#3F3F3F" :background "#2B2B2B"))))
     '(term-color-blue ((t (:foreground "#7da6ff" :background "#4C7073"))))
     '(term-color-cyan ((t (:foreground "#93E0E3" :background "#8CD0D3"))))
     '(term-color-green ((t (:foreground "#7F9F7F" :background "#9FC59F"))))
     '(term-color-magenta ((t (:foreground "#DC8CC3" :background "#CC9393"))))
     '(term-color-red ((t (:foreground "#AC7373" :background "#8C5353"))))
     '(term-color-white ((t (:foreground "#DCDCCC" :background "#656555"))))
     '(term-color-yellow ((t (:foreground "#DFAF8F" :background "#9FC59F"))))
     '(term-default-bg-color ((t (:inherit term-color-black))))
     '(term-default-fg-color ((t (:inherit term-color-white))))

     ; Org
      '(org-block ((t (:inherit (fixed-pitch shadow) :extend t :background "gray13"))))
      '(org-ellipsis ((t (:foreground "white smoke"))))

      ; Elfeed
       '(elfeed-search-feed-face ((t (:foreground "dark salmon"))))
       '(elfeed-search-tag-face ((t (:foreground "gray" :height 0.8))))
       '(elfeed-search-title-face ((t (:foreground "gray55" :underline "gray25" :height 1.3))))
       '(highlight ((t (:background "gray40"))))
       '(hl-line ((t (:inherit highlight :extend t :background "gray25" :foreground "medium sea green"))))

       
     )))

